### ISSUE TITLE
SELECTED CELL update: Report card on the right panel.

### ISSUE THREAD

**Issue by egeozin** [civic-data-design-lab/atlas-lighting#99]

**Comment by thuytle** [comment]
pop up stays on if mouse hover over side bar 
<img width="1440" alt="screen shot 2017-02-27 at 12 21 21 pm" src="https://cloud.githubusercontent.com/assets/1759051/23371800/5587cd74-fce7-11e6-82b9-513993a3cc0a.png">

**Comment by cosmicespresso** [comment]
Create a report card on the right which shows values when the cell is selected.

- [x] Create second tab

- [x] Add names to both tabs

- [x] jQuery transitions upon click

- [x] When a cell is selected then the color of the tab should turn to blue (or flash to blue)

- [x] Use the metric order from the left hand panel

- [x] Move the Instagram thumbnails to Selected Cell View

- [x] Move the Google Street View to Selected Cell View

- [x] Remove/hide from left side

- [x] for business types and instagram content topics, it can be a string of top 3 types or topics. Example: Health, Retail, Service

- [x] dataset name could be renamed as Top 3 Business Types in this Cell

- [x] only works when the URL is http://localhost:8080/view/grid.html#Chicago*street_view|instagram_pics  --> change the jQuery

- [ ] google streetview is hidden under an overlay?

**Comment by hgunc** [comment]
in report card view, we'll have to keep the instagram thumbnails and street view as it is.

Also for business types and instagram content topics, it can be a string of top 3 types or topics.

Example: Health, Retail, Service 

And the dataset name could be renamed as Top 3 Businees Types in this Cell

**Comment by cosmicespresso** [comment]
@egeozin did you move parts of the jQuery controlling the behavior out of the `grid_map.js`? Can't find the parts that control the click and fold events on the left and right panels.

**Comment by egeozin** [comment]
Yep, I put them under control.js. Sorry, I put a notice here two weeks ago and then closed it.

**Comment by cosmicespresso** [comment]
Cool, I found them and am almost done with this issue.

**Questions**

1. What is the status of the `topicsData`?

2. Should we entirely move the Google Street view and the Instagram Thumbnails to the "Cell View" tab?

**Comment by egeozin** [comment]
What do you mean? Do you mean the script running on the linux machine? Or do you mean whether we have it for LA and Chicago? (We are waiting for the merged csv file(with topics) for LA. As for the Chicago, we are waiting for the script to end in the compute engine(that will output aggregate topics), it shows 3 days)

Another question: have you updated the pittsburgh_grid.csv under the data folder recently?

**Comment by cosmicespresso** [comment]
Oh right, I completely forgot we were still waiting for that stuff to compile. Ok so for the moment I don't show anything for the Instagram topics in the CELL VIEW.

Regarding your question: Yes I changed the instagram likes to the correct naming (I'm pretty sure thats what I did at least). 

How about the second question above? (Sorry I just modified my post so you might have missed it). I was thinking that the Google Street view and the Insta thumbnails are always cell-specific, so it makes sense to show them in that separate tab? 🤔

**Comment by egeozin** [comment]
Your response to the second question is great! Let's do that, we do not need to show them in the map view at all. 

Another thing: have you noticed the small disconnect in the left border line of the map - view, I will open another thread regarding whether we should make that "drag a dataset to upload" box, but before that it may be due to recent changes in css.

**Comment by egeozin** [comment]
With your response, we are also circumventing the problem of incoherent google street view api problem(It sometimes shows sometimes not due to connection with google's api ). This way it will not be distracting in the map view tab. 

By the way can you try panning in the map? You are selecting the cells don't you? If not, I will continue to wonder why do I always select a cell whenever I pan :)

**Comment by cosmicespresso** [comment]
1. Panning
Yes the UX is not great - but it gets better once I zoom in a little bit. The way you deal with panning is probably by being careful with the jQuery events, and/or a Mapbox panning property. Not sure we have the time for that...

2. Google street + Instagram thumbnails
Great, I will incorporate them in the CELL VIEW.
Do you think that we should just take them out of the left hand side then? As in, they automatically show up only when we have selected a single cell, and they can not be dragged and dropped off. Sounds like a big change but makes sense to me. Maybe we check with Hayro?

3. Left border disconnect
I know, and it bothers me, will try to fix it later :)
![image](https://cloud.githubusercontent.com/assets/10440556/23445053/b6969b8e-fe06-11e6-946b-f074583a066c.png)

**Comment by egeozin** [comment]
1. It's no big deal, I may try to enhance it if it becomes intolerable(highly unlikely).

2. Wonderful, let's do that, we can take it back according to feedback.

3. Don't bother, I can look into that, after you finish your css changes.

**Comment by cosmicespresso** [comment]
Here is the state so far:

![image](https://cloud.githubusercontent.com/assets/10440556/23475009/f3fcc692-fe83-11e6-9379-bfe6d77added.png)

@hgunc what do you think of the design?

Still need to be addressed:

- [ ] Instagram topics data

- [ ] Google street view does not work reliably. It either will work for every cell, or show a black screen for every cell (see below). 

@egeozin any ideas about that? I see some errors in the console but they seem to be from Instagram photos not properly loading from Firebase. 

![image](https://cloud.githubusercontent.com/assets/10440556/23475088/4e8bc342-fe84-11e6-8c54-276c99f848fa.png)

**Comment by egeozin** [comment]
Yep, these two errors you mentioned looks like they are related with Google's external and internal API calls. You make some refreshes and hope for the best, I have tried to fix it but I see similar complains around the web, let me look what to do about it. 

In the instagram thumbnails case, those errors are due to the output of callback we are receiving from the Google API. So when a user deletes an image, the api asks with an obsolete link.  I can try to find a way to hide the callback error print.

**Comment by egeozin** [comment]
Now, I cannot see instagram thumbnails for LA, have you changed anything?

**Comment by cosmicespresso** [comment]
No I haven't, I also don't see them. Can you check something - I told Dennis but he not sure if he updated anything: I have noticed that the formats of `LA.csv` and `LA_grid.csv` seem to not look like the rest of the `[city].csv` and `[city]_grid.csv` files. In fact they seem to be mixed up (the LA.csv looks like the Chicago_grid.csv for example). Maybe this has also bled into how the Firebase database is hit? Not sure why LA only has this issue.


Update: 
![image](https://cloud.githubusercontent.com/assets/10440556/23484726/26b412b2-fea6-11e6-8e11-7c095feaa0a1.png)

?? It showed up after 2 minutes.

**Comment by egeozin** [comment]
Yeah it shows up, but no images inside, it's probably related to the firebase. I just checked the firebase and I cannot access the database except a loading spinner(Hmm, site related problems I guess). So, yesterday, I changed the api query for other cities, as the database containing links to images for those cities have a different format than LA's. However, I triple-checked that I was able to see the images for LA(because I kept the query for LA, I only change the query when other cities are selected, so it's not related with our script).

So for the strangeness in LA_grid.csv file, it's because there are no fields with "insta_{number}" in LA. Also, The reason for undefined insta_likes in LA is that there is no insta_like field. I will double check with Dennis, I assume an update on it. 

We are not using the csv's with no "grid" in it, I will also get rid of them soon, as there are lots of unused junk out there.
